libs/cart/src/drivers/magento/transforms/outputs/cart-payment.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+
+import { DaffCartPaymentMethod } from '../../../../models/cart-payment';
+import { MagentoCartPaymentMethod as MagentoCartPayment } from '../../models/outputs/cart-payment-method';
+
+/**
+ * Transforms magento carts into an object usable by daffodil.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartPaymentTransformer {
+
+  /**
+   * Transforms the magento CartPayment from the magento cart query into a DaffCartPaymentMethod.
+   * @param response the response from a magento cart query.
+   */
+  transform(responsePayment: MagentoCartPayment): DaffCartPaymentMethod {
```

Comment:
This implies that what we get back is only a `DaffCartPaymentMethod`. We can be covariant here and return additional information.

Reply:
Should this be a generic on the driver? `T extends DaffCartPaymentMethod`

Reply:
I believe the driver is already generic.

Reply:
Sorry I meant transformer. I think it will be sufficient to just remove the return type though.

Reply:
Changed in [4c352f5](https://github.com/graycoreio/daffodil/pull/654/commits/4c352f543a6ff12de8b7c762c5f1b0c5da8f7299).